### PR TITLE
[Major Refactor] Native typescript support, circular references fix, vitest

### PR DESCRIPTION
**Pull request by jonluca**
This diff:

* Rewrites the core logic in typescript
* Creates separate CJS and ESM exports
* Brings in the `cloudflare/json-schema-tools` because that library seems to have become abandonware, and has some small bugs in it that haven't been fixed
* Changes the JSON dereferencer to use the more actively maintained `json-schema-ref-parser`
* Adds a new option to prevent cyclic referenes
* Changes the tests to use vitest
* Adds a new test for circular references

Fixes
- #27
- #1 
- #4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,36 +1,57 @@
...
-    "@cloudflare/json-schema-walker": "^0.1.1",
-    "@stoplight/json-ref-resolver": "^3.1.4",
-    "@stoplight/yaml": "^4.2.3",
-    "chalk": "^5.0.1",
-    "node-fetch": "^2.6.0",
-    "tslib": "^2.4.0",
-    "yargs": "^17.5.1"
-  }
+	"name": "browser-json-schema-to-openapi-schema",
+	"version": "0.1.4",
```
Comment (philsturgeon) [inline package.json:3]: Can we change this back so autorelease will work.
```suggestion
	"name": "@openapi-contrib/json-schema-to-openapi-schema",
	"version": "0.0.0-development",
```

**Comment by philsturgeon** [comment]
I see https://github.com/jonluca/json-schema-walker/ over here but how is that ending up in NPM.

**Comment by jonluca** [comment]
Thanks!

Yeah let me change the name and version.

> and I [cannot find that on npmjs.org](https://www.npmjs.com/search?q=json-schema-walker).

Is https://www.npmjs.com/package/json-schema-walker not visible? I didn't think it was private - I'm using it in this package and it seems to be working

>  how is that ending up in NPM.

I manually published it

**Review by philsturgeon — APPROVED** [review]

**Comment by philsturgeon** [comment]
Amazing, thank you. If you ever have a bit of time in your hands, the sister package could do with Similar love as the testing and JavaScript is all super old school. 

https://github.com/openapi-contrib/openapi-schema-to-json-schema

Understandable if not, I’ll have a go sometime but I’m spending the next few weeks watering saplings so our freshly planted woodlands don’t die and that means less time for code.

**Comment by jonluca** [comment]
Cool, sure https://github.com/openapi-contrib/openapi-schema-to-json-schema/pull/45

**Pull request merged**